src/server/serve.rs

```diff
@@ -355,13 +411,14 @@ impl InnerService {
                 if let Some(range) = req.headers().typed_get::<Range>() {
                     #[allow(clippy::single_match)]
                     match (
-                        is_range_fresh(req, &etag, &last_modified),
+                        is_range_fresh(&req, &etag, &last_modified),
                         is_satisfiable_range(&range, size as u64),
                     ) {
                         (true, Some(content_range)) => {
                             // 206 Partial Content.
                             if let Some(range) = content_range.bytes_range() {
-                                body = send_file_with_range(&path, range);
+                                let stream = send_file_with_range(&path, range)?;
+                                body = Body::wrap_stream(stream);
```

Comment:
Given the content-length of range request is fixed, we can easily add it (if not compressed).

Reply:
set Content-Length for partial content request on 481b1fb